src/libs/ReportUtils.js

```diff
@@ -411,16 +412,73 @@ function formatReportLastMessageText(lastMessageText) {
 }
 
 /**
- * Helper method to return a default avatar
+ * Helper method to return the default avatar associated with login
  *
  * @param {String} [login]
  * @returns {String}
  */
 function getDefaultAvatar(login = '') {
-    // There are 8 possible default avatars, so we choose which one this user has based
+    // There are 24 possible default avatars, so we choose which one this user has based
     // on a simple hash of their login
+    if (!login) {
+        return Expensicons.FallbackAvatar;
+    }
+    if (login === CONST.EMAIL.CONCIERGE) {
+        return Expensicons.ConciergeAvatar;
+    }
+
+    const loginHashBucket = (Math.abs(hashCode(login.toLowerCase())) % 24) + 1;
+
+    return defaultAvatars[`Avatar${loginHashBucket}`];
+}
+
+/**
+ * Helper method to return old dot default avatar associated with login
+ *
+ * @param {String} [login]
+ * @returns {String}
+ */
+function getOldDotDefaultAvatar(login = '') {
+    // There are 8 possible old dot default avatars, so we choose which one this user has based
+    // on a simple hash of their login
+    if (login === CONST.EMAIL.CONCIERGE) {
+        return CONST.CONCIERGE_ICON_URL;
+    }
     const loginHashBucket = (Math.abs(hashCode(login.toLowerCase())) % 8) + 1;
-    return `${CONST.CLOUDFRONT_URL}/images/avatars/avatar_${loginHashBucket}.png`;
+
+    return `${CONST.CLOUDFRONT_URL}/images/avatars/avatar_${loginHashBucket}`;
+}
+
+/**
+ * Returns true if provided URL requires a default avatar
+ * @param {String} [avatarURL]
+ * @returns {Boolean}
+ */
+function isDefaultAvatar(avatarURL) {
+    if (isString(avatarURL) && (avatarURL.includes('images/avatars/avatar_') || (avatarURL.includes('images/avatars/user/default')))) {
+        return true;
+    }
+
+    // if null URL, we should also use a default avatar
+    if (!avatarURL) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Provided a source URL, if source is a default avatar, return the associated SVG.
+ * Otherwise, return the URL pointing to a user-uploaded avatar.
+ *
+ * @param {String} [avatarURL] - the avatar source from user's NVPs
+ * @param {String} [login] - the email of the user
+ * @returns {String | Function}
+ */
+function getCorrectAvatar(avatarURL, login) {
```

Comment:
Just riffing... I know we passed `login` to `getDefaultAvatar` in the past so I see why you kept that for all of these functions, but... What if we instead just use `sessionEmail` which already exists at the top of this file? Do you think there's any downside in using that? It could clean up a decent amount of logic where we call these functions

Reply:
Ok now I'm realizing that we actually need to be able to pass `login` for cases where we're getting the correct avatar for people who aren't the main user - but maybe for the other case (where we want to get the user's avatar) we don't pass anything, and default to `sessionEmail`? Just wondering what your thoughts are, maybe this is all unnecessary what I'm talking about haha

Reply:
My initial thoughts about not having login default was that we could then use the fallback avatar more meaningfully. In theory, we should have at least the login for every user we get an avatar for, so it prevents bugs like mismatched avatars caused by falsely thinking you've passed in the proper login

Reply:
I think you're right, it's better to have to pass an email every time so we don't accidentally forget some places where we need to, or vise versa - resolving!